packages/version/src/__tests__/version-bump.spec.ts

```diff
@@ -124,6 +124,26 @@ describe('version bump', () => {
     expect(message).toContain('package-5@5.0.1-alpha.0');
   });
 
+  test('prerelease increments version for the filtered packages when providing a --scope', async () => {
+    const testDir = await initFixture('independent');
+
+    // await new VersionCommand(createArgv(testDir, "--bump", "prerelease"));
+    await factory(createArgv(testDir, '--bump', 'prerelease', '--scope', 'package-1'));
+
+    const message = await getCommitMessage(testDir);
+    expect(message).toContain('package-1@1.0.1-alpha.0');
```

Comment:
the `message` here is different compare to the Publish tests, this `message` is a string (or a string array, I forgot) that will contain (or not) all the packages. So yes you did check the pkg-1 **but** you should also check that the other packages are **not** included in that `message`, that is basically what I wrote and meant in my previous comment

> and then the last expect should be an inverse to make sure it's filtered out after the scope
> 
> ```ts
>  expect(message).not.toContain('package-5@5.0.1-alpha.0');
> ```

Reply:
@ghiscoding sorry i missed that

Reply:
btw i think there is an issue with those tests. If i did the ignore test with package 5 then the toContain was not working. Feels like the `message` passed to `expect` is truncated or something. I had to use package 1 or i could not get it to work

Reply:
you can add a breakpoint to your test and see the content of `message` then adjust your test according to it

Reply:
@ghiscoding i did the debugger was showing the full text with line breaks but the `.toContain` was not working

Like this is not working when i think it should:
```ts
test('prerelease increments version only for the packages that are not --ignore(d)', async () => {
    const testDir = await initFixture('independent');

    // await new VersionCommand(createArgv(testDir, "--bump", "prerelease"));
    await factory(createArgv(testDir, '--bump', 'prerelease', '--ignore', 'package-@(2|3|4|5|1)'));

    const message = await getCommitMessage(testDir);
    expect(message).toContain('package-6@1.0.1-alpha.0');
  });

```

![image](https://github.com/user-attachments/assets/4adc5317-3876-43e9-beed-5d607d205d1b)

Reply:
EDIT: my bad i see now that versions were not the same for all packages... i misread the version numbers!
All good